internal/repository/verify.go

```diff
@@ -22,47 +22,61 @@ import (
 // another is to create a new RSL entry for the current state.
 var ErrRefStateDoesNotMatchRSL = errors.New("Git reference's current state does not match latest RSL entry") //nolint:stylecheck
 
-func (r *Repository) VerifyRef(ctx context.Context, target string, latestOnly bool) error {
+func (r *Repository) VerifyRef(ctx context.Context, localRefName, remoteRefName string, latestOnly bool) error {
```

Comment:
nit: IIUC, this isn't really a remote ref, even though it's most likely going to be used in support of remote refs?

i.e. This is really verifying a local ref with a name override. This doesn't actually interact with a remote in any way. If this does interact with a remote, how do we know which remote to work with?

Reply:
yes, it doesn't interact with the remote, this is about the name for the ref used on the remote, which maps to the name used in the RSL entry for pushes to that ref. I could rename this as a nameOverride or so?

Reply:
I've renamed this to be refName and refNameOverride and changed the logic so we actually _use_ refName throughout, while using the override to update the refName value if needed.

Reply:
I've also changed the debug message to be a bit clearer.